### PR TITLE
Upgrade babel code frame

### PR DESCRIPTION
**Pull request by SimenB**
New version takes a loc object directly. babel/babel#5646.
I messed up a `function` keyword for the purposes of this screenshot.

image
image

/cc @hzoo seems like it's consistent with old version to be off by one. But might be worth fixing as it seems most parsers has to get the column shifted, which makes it sorta painful to use...

### CODE REVIEW

File: src/parser-flow.js [hunk 1]
```diff
@@ -13,11 +13,11 @@ function parse(text) {
   });
 
   if (ast.errors.length > 0) {
-    throw createError(
-      ast.errors[0].message,
-      ast.errors[0].loc.start.line,
-      ast.errors[0].loc.start.column + 1
-    );
+    const loc = ast.errors[0].loc;
+    throw createError(ast.errors[0].message, {
```
Comment (SimenB) [inline src/parser-flow.js]: here it would be awesome to just pass in `loc`, instead of having to `++` it

File: src/parser-postcss.js [hunk 2]
```diff
@@ -174,11 +174,12 @@ function parseNestedCSS(node) {
       } catch (e) {
         const line = +(e.toString().match(/line: ([0-9]+)/) || [1, 1])[1];
         const column = +(e.toString().match(/column ([0-9]+)/) || [0, 0])[1];
-        throw createError(
-          "(postcss-values-parser) " + e.toString(),
-          node.source.start.line + line - 1,
-          node.source.start.column + column + node.prop.length
-        );
+        throw createError("(postcss-values-parser) " + e.toString(), {
```
Comment (SimenB) [inline src/parser-postcss.js]: I was unable to test that this is correct. How do I trigger this?

Comment (vjeux) [inline src/parser-postcss.js]:
```js
a { 
  color: a);
}

SyntaxError: (postcss-values-parser) Error: Unbalanced parenthesis (2:8)
  1 | a { 
> 2 |   color: a);
    |        ^
  3 | }
```

Comment (vjeux) [inline src/parser-postcss.js]: Looks like it's off by one or two already :p

**Comment by vjeux** [comment]
So cool to have range support!

**Comment by hzoo** [comment]
If it's always +1 we can take a look into fixing that!

**Comment by SimenB** [comment]
@vjeux Can you tell me how the weird line and column string lookup thingy came to be?

I pushed a commit removing it now, and it looks good on the example you provided:

image

**Comment by vjeux** [comment]
I've tried to point at the beginning of the value: `a)`. But it was a hacky heuristic. Displaying the entire node like this is better imo :)

**Comment by vjeux** [comment]
Let me know when I can press the merge button, it looks good to me.

File: src/parser-postcss.js [hunk 3]
```diff
@@ -27,9 +27,10 @@ function parseValueNodes(nodes) {
   const commaGroupStack = [commaGroup];
 
   for (let i = 0; i < nodes.length; ++i) {
-    if (nodes[i].type === "paren" && nodes[i].value === "(") {
+    const node = nodes[i];
```
Comment (SimenB) [inline src/parser-postcss.js]: Hit having to look into the array when debugging. Easier when it's referenced like this. Kept the change, cuz why not 🙂

**Comment by SimenB** [comment]
@vjeux This is good to go now 😄

@hzoo It's off by one in flow, babylon and typescript parser. _Not_ in postcss parser though, but I'm thinking we should focus on the JS ones.

**Pull request merged**